jetcd-core/src/main/java/io/etcd/jetcd/resolver/IPNameResolver.java

```diff
@@ -61,21 +66,33 @@ public IPNameResolver(URI targetUri) {
         this.lock = new Object();
         this.targetUri = targetUri;
         this.authority = targetUri.getAuthority() != null ? targetUri.getAuthority() : SCHEME;
-        this.addresses = Stream.of(targetUri.getPath().split(","))
-            .map(address -> {
-                return address.startsWith("/")
-                    ? address.substring(1)
-                    : address;
-            })
-            .map(address -> {
+
+        List<String> defaultAuthorities = Arrays.stream(this.authority.split(",")).collect(Collectors.toList());
+        List<String> addressStrList = Stream.of(targetUri.getPath().split(","))
+            .map(
+                address -> {
+                    return address.startsWith("/") ? address.substring(1) : address;
+                })
+            .collect(Collectors.toList());
+        final boolean needOverrideAuthority = addressStrList.size() == defaultAuthorities.size();
```

Comment:
there should be some check here as id the size is not the same, it could be because of a misconfiguration